packages/kit/src/runtime/client/client.js

```diff
@@ -404,6 +394,17 @@ export function create_client(app, target) {
 					scrollTo(0, 0);
 				}
 			}
+
+			const changed_focus =
+				// reset focus only if any manual focus management didn't override it
+				document.activeElement !== activeElement &&
+				// also refocus when activeElement is body already because the
+				// focus event might not have been fired on it yet
+				document.activeElement !== document.body;
+
+			if (!keepfocus && !changed_focus) {
+				await reset_focus();
```

Comment:
this probably was introduced through this `await` which wasn't there before, but was needed to introduce to fix a focus bug